grid/grid_network.py

```diff
@@ -34,16 +35,72 @@ def search(self, *query):
         # Connect with grid nodes that contains the dataset and get their pointers
         tensor_set = []
         for node_id, node_url in match_nodes:
-            if node_id not in sy.hook.local_worker._known_workers:
-                worker = WebsocketGridClient(sy.hook, node_url, node_id)
-                worker.connect()
-            else:
-                # There is already a connection to this node
-                worker = sy.hook.local_worker._known_workers[node_id]
-                worker.connect()
+            worker = self.__connect_with_node(node_id, node_url)
             tensor_set.append(worker.search(*query))
         return tensor_set
 
+    def host_model(self, model, model_id):
```

Comment:
can you call this serve_model instead? This will imply changing the notebook!

Reply:
Done! (fa2ea61)

Reply:
Done! (fa9c9bb)